src/handle_action.cpp

```diff
@@ -787,6 +787,10 @@ static void smash()
             return;
         }
     }
+
+    const int bash_furn = m.furn( smashp ).obj().bash.str_min;
+    const int bash_ter = m.ter( smashp ).obj().bash.str_min;
```

Comment:
I think this works, and is preferred.
```suggestion
    const int bash_ter = m.ter( smashp )->.bash.str_min;
```